src/services/redis.ts

```diff
@@ -88,13 +91,23 @@ export class RedisService {
    * @returns {Promise<string[]>} A Promise that resolves to an array of set members as strings.
    */
   async getMembers(key: string): Promise<string[]> {
+    const _serialiseRuleResult = fastJson({
+      title: 'FastJson Schema',
+      ...messageSchema.definitions,
+    });
+
     try {
-      const res = await this._redisClient.smembers(key);
-      if (!res || res.length === 0) {
+      const res = (await this._redisClient.smembersBuffer(key)) as Uint8Array[];
+      const membersBuffer = res.map((member) => {
+        const decodedMember = FRMSMessage.decode(member);
+        return _serialiseRuleResult(FRMSMessage.toObject(decodedMember));
```

Comment:
Why not this?:

```suggestion
        return decodedMember.toJSON();
```

This will convert to an Object first and then will convert to JSON, which is eventually deserialised back to an object by this method's caller.
```js
_serialiseRuleResult(FRMSMessage.toObject(decodedMember));
```

If we just call `.toJSON()` then we skip the roundabout?